### ISSUE TITLE
Moving to .NET 4.6

### ISSUE THREAD

**Issue by touil** [jmichelp/sdrsharp-bladerf#3]
SDR# has moved to .NET 4.6 and can run on Mono 4.0 and up. Can you update the project and the release bins accordingly?

**Comment by jmichelp** [comment]
That's great news! Thank you for notifying me.

Quite a busy week so I'm not sure I probably won't be able to work on it before the week-end.

I need to update the underlying lib, ensure that it's still working as intended, deprecate old code and probably tweak it so that it also works on Linux and OS X too.

**Comment by LazyDodo** [comment]
Not many changes are needed, the marshaling of the version string goes badly for unknown reasons (by all logic it should work, it just doesn't and makes things crash), adding this didn't work

```
        [MarshalAs(UnmanagedType.LPStr)]
        public string describe;
```

manually marshaling works though. 

```
  public struct bladerf_version
    {
        public UInt16 major;
        public UInt16 minor;
        public UInt16 patch;
        public IntPtr describePtr;
        public string describe
        {
            get
            {
                return Marshal.PtrToStringAnsi(describePtr);
            }
        }
    }
```

Other than that there's an issue with the mainform not responding well to the samplerate changed events, but i send touil a fix for that.

**Comment by jmichelp** [comment]
Unfortunately I didn't have time to work on it. Thanks a lot LazyDodo for that. I applied your fixes.

Hopefully next week I will have more time to work on a new release. I also saw a while ago in the changelog new features on SDRSharp (center locking, etc.). Do I have some new interfaces to implement on the frontend side?

I will also update the documentation as soon as the release of the plugin is available and tested.
